Lua/tests/syntax_test_lua.lua

```diff
@@ -0,0 +1,491 @@
+
+    {[a] = x, b = y};
+--   ^^^ meta.brackets
+--   ^ punctuation.section.brackets.begin
+--    ^ variable.other
+--     ^ punctuation.section.brackets.end
+--       ^ punctuation.separator.key-value
+--         ^ variable.other
+--          ^ punctuation.separator.field
+--            ^ meta.key
+--              ^ punctuation.separator.key-value
+--                ^ meta.mapping variable.other
+
+--PARENS
+
+    (foo + bar);
+--  ^^^^^^^^^^^ meta.group
+--  ^ punctuation.section.group.begin
+--            ^ punctuation.section.group.end
+
+    foo.bar;
+--     ^ punctuation.accessor
+--      ^^^ meta.property
+
+    foo:bar;
+--     ^ punctuation.accessor
+--      ^^^ meta.property
+
+    foo.baz();
+--     ^ punctuation.accessor
+--      ^^^ meta.property variable.function
+
+    foo.baz "";
+--     ^ punctuation.accessor
+--      ^^^ meta.property variable.function
+
+    foo.baz '';
+--     ^ punctuation.accessor
+--      ^^^ meta.property variable.function
+
+    foo.baz [=[ a string ]=];
+--     ^ punctuation.accessor
+--      ^^^ meta.property variable.function
+
+    foo.baz {};
+--     ^ punctuation.accessor
+--      ^^^ meta.property variable.function
+--          ^^ meta.function-call.arguments meta.mapping
+
+    foo[bar + baz];
+--     ^^^^^^^^^^^ meta.brackets
+--     ^ punctuation.section.brackets.begin
+--               ^ punctuation.section.brackets.end
+
+--FUNCTION CALLS
+
+    f(42);
+--  ^ variable.function
+--   ^ meta.function-call.arguments meta.group
+--   ^ punctuation.section.group.begin
+--      ^ punctuation.section.group.end
+
+    f "argument";
+--  ^ variable.function
+--    ^^^^^^^^^^ meta.function-call.arguments string.quoted.double
+
+    f
+    'argument';
+--  ^^^^^^^^^^ meta.function-call.arguments string.quoted.single
+
+    f [[ foo ]];
+--  ^ variable.function
+--    ^^^^^^^^^ meta.function-call.arguments string.quoted.block
+
+    f {};
+--  ^ variable.function
+--    ^^ meta.function-call.arguments meta.mapping
+
+--FUNCTIONS
+
+    function function_name( a, b, ... )
+--  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.function
+--  ^^^^^^^^ storage.type.function
+--           ^^^^^^^^^^^^^ entity.name.function
+--                        ^^^^^^^^^^^^^ meta.group
+--                        ^ punctuation.section.group.begin
+--                          ^ variable.parameter.function
+--                           ^ punctuation.separator.comma
+--                             ^ variable.parameter.function
+--                              ^ punctuation.separator.comma
+--                                ^^^ constant.language
+--                                    ^ punctuation.section.group.end
+    end
+--  ^^^ meta.function meta.block keyword.control.end
+
+    function foo.bar:baz (...) end
+--  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.function
+--           ^^^ variable.other - entity
+--              ^ punctuation.accessor
+--               ^^^ variable.other - entity
```

Comment:
This should be a `meta.property`, not `variable.other`.

Works correctly for `foo.abc.bar = function() end;`, although there is no test for it.

Reply:
Fixed.